app/src/main/java/de/grobox/transportr/trips/search/DirectionsViewModel.kt

```diff
@@ -121,6 +121,7 @@ class DirectionsViewModel @Inject internal constructor(
     fun setProducts(newProducts: EnumSet<Product>) {
         _products.value = newProducts
         search()
+        settingsManager.setPreferredProducts(newProducts)
```

Comment:
I don't think it's a good idea to update the settings from within the UI thread. Maybe double-check how it is done for other settings?

Reply:
Totally agree with you on this. That said, from what I see, all `SettingsManager` methods are currently called on the main thread. I will create an issue to improve this.

Reply:
Can we leave this as-is?
This shouldn't block the UI thread and, from what I see, other methods in `SettingsManager` are called from the main thread too.

Reply:
Alright, that's good to know! We can leave it as-is in that case I'd say!